lib/msf/base/config.rb

```diff
@@ -50,6 +50,24 @@ def self.get_config_root
     end
   end
 
+  # Determines the git hash for this source tree
+  #
+  # @return [String] the git hash for this source tree
+  def self.get_git_hash
+    msfrelease = File.join(InstallRoot, 'msfrelease.txt')
+    hash = 'unknown'
+    @git_hash ||= begin
+      if File.exist?(msfrelease)
+        hash = File.read(msfrelease)
+      else
+        if File.exist?(File.join(InstallRoot, '.git'))
+          hash = `git rev-parse HEAD`
```

Comment:
Oh you want me to propose SOLUTIONS rather than just point out problems? Damn, called my bluff ;)

Yeah you make a good point re: gemspec. I'm honestly not sure what would be considered the right thing here. I do think that trying to find git would suck.

If git isn't found would it result in the banner containing an error?